src/Integrations/Integrations/LaravelQueue/LaravelQueueIntegration.php

```diff
@@ -0,0 +1,330 @@
+                    $exception,
+                    $args[2] ?? null,
+                    get_class($this)
+                );
+            }
+        );
+
+        // Laravel 8
+        trace_method(
+            'Illuminate\Bus\Batch',
+            'add',
+            function (SpanData $span, $args, $retval, $exception) use ($integration) {
+                $integration->setSpanAttributes(
+                    $span,
+                    'laravel.queue.batch.add',
+                    'send',
+                    null,
+                    $exception,
+                    null,
+                    get_class($this)
+                );
+                if ($retval) {
+                    $span->meta['messaging.laravel.batch_id'] = $retval->id;
+                }
+            }
+        );
+
+        return Integration::LOADED;
+    }
+
+    public function setSpanAttributes(
+        SpanData $span,
+        string $name,
+        $operation = null,
+        $job = null,
+        $exception = null,
+        $queue = null,
+        $resourceSubstitute = null
+    ) {
+        $span->name = $name;
+        $span->service = $this->getAppName();
+        $span->type = 'queue';
+        $span->meta[Tag::SPAN_KIND] = 'client';
+        $span->meta[Tag::COMPONENT] = LaravelQueueIntegration::NAME;
+
+        if ($operation) {
+            $span->meta[Tag::MQ_OPERATION] = $operation;
+        }
+
+        if ($job instanceof Job) {
+            $jobName = $job->resolveName();
+            $span->meta = array_merge(
+                $span->meta,
+                $this->getMetadataFromJob($job)
+            );
+            $queue = $queue ?? $job->getQueue();
+        } elseif (is_object($job)) { // Most certainly a CallQueuedClosure
+            $jobName = method_exists($job, 'displayName') ? $job->displayName() : get_class($job);
+            $connectionName = $job->connection ?? config('queue.default');
+            $queue = $queue ?? ($job->queue ?? (config("queue.connections.$connectionName.queue") ?? 'default'));
+
+            $span->meta['messaging.laravel.name'] = $jobName;
+            $span->meta['messaging.laravel.queue'] = $queue;
+            $span->meta['messaging.laravel.connection'] = $connectionName;
+            $span->meta = array_merge(
+                $span->meta,
+                $this->getMetadataFromObject($job)
+            );
+        } else {
+            $jobName = $job;
+        }
+
+        if ($resourceSubstitute) {
+            $span->resource = $resourceSubstitute;
+        } else {
+            $span->resource = $queue ? "$jobName -> $queue" : "$jobName";
+        }
+
+        if ($exception) {
+            $this->setError($span, $exception);
+        }
+    }
+
+    public function getMetadataFromJob(Job $job)
+    {
+        $metadata = [
+            'messaging.laravel.attempts' => $job->attempts(),
+            'messaging.laravel.connection' => $job->getConnectionName() ?? config('queue.default'),
```

Comment:
The connection name will be something like 'database', 'sync', or 'redis' (non-exhaustive, these are defined in the `config/queue.php` file). It doesn't really have the format we would expect from a url in my opinion. However, if you feel like this makes sense, then that's ok for me.

Reply:
is this like this? `'mysql:host=mysql_integration;dbname=test'` 
In those cases my concerns are more about leaking sensitive info?

Reply:
Laravel connections are defined using an associative array ([as can be seen in this sample configuration](https://github.com/agence-webup/laravel-skeleton/blob/2f93dfd40ede154361cc2a9fd3f2000560e071fc/config/queue.php#L31)) with specific **configuration keys**, rather than a single string containing all parameters. Therefore, the value of `messaging.laravel.connection` would be one of these **configuration keys**, referencing to one of the defined queue connection types. 

Note: A 'connection' in the context of laravel queue refers to the method by which the queue system communicates with the underlying storage mechanism used to store and retrieve queued jobs.